src/edu/psu/sweng500/team8/gui/BlockGUI.java

```diff
@@ -0,0 +1,164 @@
+//		cells[1][1].setMinimumSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
+//		cells[1][1].setPreferredSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 1;
+		gridBagConstraints.gridy = 1;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+		add(cells[1][1], gridBagConstraints);
+
+		cells[1][2] = new CellGUI();
+//		cells[1][2].setMaximumSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
+//		cells[1][2].setMinimumSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
+//		cells[1][2].setPreferredSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 1;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 1);
+		add(cells[1][2], gridBagConstraints);
+
+		cells[2][0] = new CellGUI();
+//		cells[2][0].setMaximumSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
+//		cells[2][0].setMinimumSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
+//		cells[2][0].setPreferredSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 0;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 1, 1, 0);
+		add(cells[2][0], gridBagConstraints);
+
+		cells[2][1] = new CellGUI();
+//		cells[2][1].setMaximumSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
+//		cells[2][1].setMinimumSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
+//		cells[2][1].setPreferredSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 1;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 0);
+		add(cells[2][1], gridBagConstraints);
+
+		cells[2][2] = new CellGUI();
+//		cells[2][2].setMaximumSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
+//		cells[2][2].setMinimumSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
+//		cells[2][2].setPreferredSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 1);
+		add(cells[2][2], gridBagConstraints);
+	}	
+	
+	public void populate(Block block, GameSession gameSession, boolean isRefresh, FocusAdapter focusAdapter, MouseAdapter mouseAdapter) {
+		
+		cells[0][0].populate(block.getCell(0), gameSession, isRefresh, focusAdapter, mouseAdapter);

```

Comment:
Yes, that was the original intention and I shall do that.  They are spread
out because they were migrated from Windows Builder generated code. I will
update all 'loop-able' sections.

On Sun, Jul 12, 2015 at 10:05 PM, Jeremy Nelson <notifications@github.com>
wrote:

> In src/edu/psu/sweng500/team8/gui/BlockGUI.java
> <https://github.com/garnryang/Team8/pull/13#discussion_r34429473>:
>
> > +
> > +		cells[2][2] = new CellGUI();
> > +//		cells[2][2].setMaximumSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
> > +//		cells[2][2].setMinimumSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
> > +//		cells[2][2].setPreferredSize(new java.awt.Dimension(CELL_SIZE, CELL_SIZE));
> > +		gridBagConstraints = new java.awt.GridBagConstraints();
> > +		gridBagConstraints.gridx = 2;
> > +		gridBagConstraints.gridy = 2;
> > +		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
> > +//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 1);
> > +		add(cells[2][2], gridBagConstraints);
> > +	}	
> > +	
> > +	public void populate(Block block, GameSession gameSession, boolean isRefresh, FocusAdapter focusAdapter, MouseAdapter mouseAdapter) {
> > +		
> > +		cells[0][0].populate(block.getCell(0), gameSession, isRefresh, focusAdapter, mouseAdapter);
>
> I would probably write this with a loop. Row index is i/3, column is i%3.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/13/files#r34429473>.
>